configtype/time.go

```diff
@@ -0,0 +1,164 @@
+package configtype
+
+import (
+	"encoding/json"
+	"fmt"
+	"regexp"
+	"time"
+
+	"github.com/invopop/jsonschema"
+)
+
+type timeType int
+
+const (
+	timeTypeZero timeType = iota
+	timeTypeFixed
+	timeTypeRelative
```

Comment:
Do wen need this enum? Can we check if `t.duration != nil` instead to know if it's a relative time?